pkg/capiutil/common.go

```diff
@@ -0,0 +1,42 @@
+package capiutil
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capiconditions "sigs.k8s.io/cluster-api/util/conditions"
+)
+
+// WaitForCondition checks if the specified object obj passes the specified
+// condition check. If the check fails, it will sleep 1 minute and check again.
+// When the check is passed, the function just returns.
+func WaitForCondition(t *testing.T, obj TestedObject, conditionType capi.ConditionType, check ConditionCheck, objGetterFunc ObjGetterFunc) {
```

Comment:
Should we align this function so that it uses the same retry mechanism that we used in other places? That way they all would look the same.

Reply:
:+1:  Looking into this .

Reply:
You mean something like this? https://github.com/giantswarm/sonobuoy-plugin/blob/master/tests/cptcconnectivity/cptcconnectivity_test.go#L69-L92

Reply:
Yes!

Reply:
Take a look now, you meant something like that?